api/v1beta1/azuremanagedcontrolplane_types.go

```diff
@@ -327,6 +327,7 @@ const (
 // AKSSku - AKS SKU.
 type AKSSku struct {
 	// Tier - Tier of an AKS cluster.
+	// +kubebuilder:default:="Free"
```

Comment:
Generally speaking I like these changes. My only question is about how these introductions relate to the API contracts. Is there documentation anywhere that says what effect this (and ostensibly other CRD validation) has on expectations for API versioning?

More concretely: are we allowed to change these values without rev'ing the API version? (I suppose this PR itself constitutes a change, but this change is merely adopting the existing behavioral contract, which IMO is not controversial.)

cc @nojnhuh

Reply:
AFAICT I'm not sure what could be practically different here between using the kubebuilder annotations and our own code to add static defaults like this. I haven't taken a close look at the PR yet but as long as we're adding the same defaults under the same conditions (only when the field is undefined), this seems safe to me from an API contract perspective. Is there some specific scenario you're thinking of that this might be breaking or otherwise constitute an API version bump?

Reply:
I'm guided entirely by creating a documented record of "did we subject this to some scrutiny". I think this is a good change, thank you @brian-cox !

Reply:
@bryan-cox 😄

Reply:
✌️ @Brian-Cox, sorry about that :)